code/game/machinery/constructable_frame.dm

```diff
@@ -30,122 +30,121 @@
 	icon_state = "box_1"
 	state = 2
 
-/obj/machinery/constructable_frame/machine_frame
-	attackby(obj/item/P as obj, mob/user as mob)
-		if(P.crit_fail)
-			to_chat(user, "<span class='warning'>This part is faulty, you cannot add this to the machine!</span>")
-			return
-		switch(state)
-			if(1)
-				if(iscablecoil(P))
-					var/obj/item/stack/cable_coil/C = P
-					if(C.get_amount() < 5)
-						to_chat(user, "<span class='warning'>You need five lengths of cable to add them to the frame.</span>")
-						return
-					playsound(loc, 'sound/items/deconstruct.ogg', 25, 1)
-					user.visible_message("<span class='notice'>[user] starts adding cables to [src].</span>",
-					"<span class='notice'>You start adding cables to [src].</span>")
-					if(do_after(user, 20, TRUE, src, BUSY_ICON_BUILD) && state == 1)
-						if(C && istype(C) && C.use(5))
-							user.visible_message("<span class='notice'>[user] adds cables to [src].</span>",
-							"<span class='notice'>You add cables to [src].</span>")
-							state = 2
-							icon_state = "box_1"
-				else
-					if(iswrench(P))
-						playsound(src.loc, 'sound/items/ratchet.ogg', 25, 1)
-						to_chat(user, "<span class='notice'>You dismantle the frame</span>")
-						new /obj/item/stack/sheet/metal(src.loc, 5)
-						qdel(src)
-			if(2)
-				if(istype(P, /obj/item/circuitboard/machine))
-					playsound(src.loc, 'sound/items/deconstruct.ogg', 25, 1)
-					to_chat(user, "<span class='notice'>You add the circuit board to the frame.</span>")
-					var/obj/item/circuitboard/machine/circuit = P
-					if(user.transferItemToLoc(P, src))
-						icon_state = "box_2"
-						state = 3
-						components = list()
-						req_components = circuit.req_components.Copy()
-						for(var/A in circuit.req_components)
-							req_components[A] = circuit.req_components[A]
-						req_component_names = circuit.req_components.Copy()
-						for(var/A in req_components)
-							var/cp = text2path(A)
-							var/obj/ct = new cp() // have to quickly instantiate it get name
-							req_component_names[A] = ct.name
-						if(circuit.frame_desc)
-							desc = circuit.frame_desc
-						else
-							update_desc()
-						to_chat(user, desc)
+/obj/machinery/constructable_frame/machine_frame/attackby(obj/item/P as obj, mob/user as mob)
```

Comment:
Change all obj/item/P to obj/item/I in attackbys, also remove the unnecessary `as` and add `params`